### PR TITLE
general fixes for DF wrapper & Navio

### LINKED ISSUE #5334: native build error
Hello.
I'm working in Ubuntu 12.04. 
Specifically,
Linux ubuntu 3.11.0-15-generic #25~precise1-Ubuntu SMP Thu Jan 30 17:39:31 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

I got a make error. (make posix_rpi_native)
Below is the error message.

Firmware/src/drivers/navio_sysfs_pwm_out/navio_sysfs_pwm_out.cpp: In function ‘int navio_sysfs_pwm_out::initialize_mixer(const char*)’:
Firmware/src/drivers/navio_sysfs_pwm_out/navio_sysfs_pwm_out.cpp:138:98: error: no matching function for call to ‘MultirotorMixer::from_text(int (&)(uintptr_t, uint8_t, uint8_t, float&), uintptr_t, char [2048], size_t&)’
    _mixer = MultirotorMixer::from_text(mixer_control_callback, (uintptr_t)&_controls, buf, buflen);

How can I fix it?

### PR DESCRIPTION
**Pull request by bkueng**
@julianoes please review & merge

fixes  #5334

### CODE REVIEW

File: src/platforms/posix/drivers/df_ak8963_wrapper/df_ak8963_wrapper.cpp [hunk 1]
```diff
@@ -279,9 +279,9 @@ int DfAK8963Wrapper::_publish(struct mag_sensor_data &data)
 	mag_report.timestamp = hrt_absolute_time();
 
 	// TODO: remove these (or get the values)
-	mag_report.x_raw = NAN;
-	mag_report.y_raw = NAN;
-	mag_report.z_raw = NAN;
+	mag_report.x_raw = 0;
+	mag_report.y_raw = 0;
+	mag_report.z_raw = 0;
```
Comment (julianoes) [inline src/platforms/posix/drivers/df_ak8963_wrapper/df_ak8963_wrapper.cpp]: What's wrong with NAN?

Comment (bkueng) [inline src/platforms/posix/drivers/df_ak8963_wrapper/df_ak8963_wrapper.cpp]: http://stackoverflow.com/questions/10366485/problems-casting-nan-floats-to-int

Comment (julianoes) [inline src/platforms/posix/drivers/df_ak8963_wrapper/df_ak8963_wrapper.cpp]: Don't cast to int, and 0 is as wrong as -2147483648.

If you have a float that is NaN, you can at least nicely catch it with `is_nan` or `!is_finite`.

Comment (bkueng) [inline src/platforms/posix/drivers/df_ak8963_wrapper/df_ak8963_wrapper.cpp]: look at the type. `mag_report.x_raw` is int16_t

Comment (julianoes) [inline src/platforms/posix/drivers/df_ak8963_wrapper/df_ak8963_wrapper.cpp]: Aha, well that's just wrong then :).

**Comment by julianoes** [comment]
Thanks, merged.

**Pull request closed without merging**